DT-6215 - Fix duplicate unknown fare route names

nurmAV opened the pull request:
## Proposed Changes

  - Previously the route name of the first unknown fare leg was used for all unknown fare legs
  - This PR adds the computation of a proper route name for each unknown fare leg individually

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build
  - Code coverage does not decrease (unless measured incorrectly)

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

viljaminurminen-cgi committed:
fix: show correct unknown fare information for each leg

vesameskanen reviewed (commented):
Code crashes sometimes:

http://localhost:8080/reitti/Lauttasaari%2C%20kaakkoinen%2C%20Helsinki%3A%3A60.151902%2C24.891956/Lahti%2C%20Lahti%3A%3A60.976337%2C25.657594/0?time=1705576016

viljaminurminen-cgi committed:
fix: check for nulls before use when computing route name

vesameskanen committed:
chore: refactor ticket information

- remove obscure and unnecessary temp arrays for unknown fares
- do not render empty divs for formatting purposes
- fix some of bad styles

vesameskanen committed:
chore: update ticket test

vesameskanen committed:
Merge branch 'v3' into DT-6215
